Fix gpmsa discrepancy corr str indexing

dmcdougall opened the pull request:
Please merge #629 (and its dependencies) before merging this PR.

dmcdougall committed:
Fix indexing of emulator precisision in setup

We don't need to conditionally check if numOutputs > 1.  The number of
emulator precision variables is always num_svd_terms, and num_svd_terms
is numOutputs by default, so in the scalar case it will be 1.

This commit also updates the regression test solutions.

dmcdougall committed:
Add max emulator basis vectors to GPMSAOptions::print

dmcdougall committed:
Fix discrepancy precision indexing in setup

This is in the hyperprior domain setup code.  There are F (== p_delta ==
numOutputs) discrepancy precision parameters, not 1.

This commit also updates the regression test solutions.

dmcdougall committed:
Fix discrepancy precision indexing in cov matrix

Now the discrepancy piece loops over discrepancy groups (assumes numOutputs
groups each with a single element) and pulls out the correct discrepancy
precision parameter.

This commit also updates the regression test solutions.

dmcdougall committed:
Fix discrepancy correlation strength indexing

This fix just applies to the indexing in the hyperparameter domain.
Updates to the test and the corresponding regression test solutions were
also made in this commit.

The test indexing needed to be updated because it assumed there was a
single discprepancy correlation strength parameter.  In actuality, there
are two, because numOutputs*dimScenario = 2 * 1 = 2.

dmcdougall committed:
Fix discrepancy correlation strength indexing

In the lnValue method we now correctly pick out the discrepancy
correlation strength belonging to the appropriate discrepancy group.
Note: we assume there are numOutputs discrepancy groups each with one
element in them.  This assumption is invalid for the functional case.

This commit updates the regresstion test solutions.

roystgnr commented on src/gp/src/GPMSA.C:
```diff
@@ -1453,7 +1453,7 @@ GPMSAFactory<V, M>::setUpHyperpriors()
      (this->m_env,
       "",
-      dimScenario,
+      numOutputs*dimScenario,
```
Could we get "F = numOutputs" into the code, not just in the comments?  Even if it's just "const unsigned int num_discrepancy_groups = numOutputs", and then we use num_discrepancy_groups here and below, that would be very helpful when I make F multivariable-vs-functional later.

dmcdougall replied:
Done.

dmcdougall committed:
Use variable for the number of discrepancy groups

This'll just make it easier when it comes to doing the functional (or
other non-default) case.
